catroid/src/main/java/org/catrobat/catroid/ui/ProjectUploadActivity.java

```diff
@@ -263,6 +273,60 @@ private void onNextButtonClick() {
 		}
 	}
 
+	private void checkCodeforApiKey() {
+		String regex = "(<value>)[0-9a-zA-Z]{16,45}";
+		codeFile = new File(project.getDirectory(), CODE_XML_FILE_NAME);
+
+		try {
+			code = com.google.common.io.Files.toString(codeFile, Charsets.UTF_8);
+		} catch (IOException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+		final List<String> matches = new ArrayList<>();
+		if (code.contains("WebRequestBrick") || code.contains("BackgroundRequestBrick") || code.contains("LookRequestBrick")) {
+			Pattern pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+			Matcher matcher = pattern.matcher(code);
+			while (matcher.find()) {
+				matches.add(matcher.group(0));
+			}
+		}
+		if (matches.size() > 0) {
+			for (String match : matches) {
+				showApiReplacementDialog(match);
+			}
+		}
+	}
+
+	private void replaceSecret(String secrete) {
+		code = code.replaceAll(secrete, "<value>INSERT-API-KEY-HERE");
+		try {
+			FileOutputStream stream = new FileOutputStream(codeFile);
+			stream.write(code.getBytes(StandardCharsets.UTF_8));
+		} catch (IOException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+	}
+
+	private boolean showApiReplacementDialog(String value) {
+		final AlertDialog alertDialog = new AlertDialog.Builder(this)
+				.setTitle(R.string.warning)
+				.setView(R.layout.dialog_replace_api_key)
+				.setPositiveButton(getText(R.string.api_replacement_dialog_accept), (dialog, which) -> {
+					replaceSecret(value);
+				})
+				.setNegativeButton(getText(R.string.api_replacement_dialog_neutral), null)
+				.setNeutralButton(getText(R.string.cancel), (dialog, which) -> {
+					finish();
```

Comment:
dialog.dismiss() instead of finish()

Reply:
@ndrnour Actually finish() is just fine here, since the dialog hast 3 options, 
-replace key
-upload with key
-cancel -> should cancel the uploading process, otherwise, it would have the same behavior as "upload with key"